internal/jimmhttp/http.go

```diff
@@ -1,56 +1,37 @@
 // Copyright 2024 Canonical.
-
-// Package jimmhttp contains utilities for HTTP connections.
 package jimmhttp
 
 import (
 	"context"
 	"net/http"
-	"net/url"
-	"strings"
-)
 
-type contextPathKey string
-
-// PathElementFromContext returns the value of the path element previously
-// extracted in a StripPathElement handler.
-func PathElementFromContext(ctx context.Context, key string) string {
-	s, _ := ctx.Value(contextPathKey(key)).(string)
-	return s
-}
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+)
 
-// StripPathElement returns a handler that serves HTTP requests by removing
-// the first element from the request path and invoking the handler h.
-//
-// If a key is specified the removed element will be stored in the context
-// attached to the request such that it can be retrieved using
-// PathElementFromContext.
-//
-// If the request URL contains a RawPath field then it will be cleared
-// before calling h.
-func StripPathElement(key string, h http.Handler) http.Handler {
```

Comment:
I can see that these have just been moved to utils.go but I'd suggest doing mechanical/moving changes in a specific commit or separate PR to make it easier to see the changes.

Reply:
for some reason I though the file rename would have been handled by git.
Guess I broke the algorithm by renaming another file with the previous name